demo-circuit/src/naive_threshold_sig_w_key_rotation/tests/mod.rs

```diff
@@ -0,0 +1,1340 @@
+    let num_custom_fields = 1;
+    let valid_sigs;
+    let threshold;
+    let is_first_certificate: bool;
+    let mut change_signing_keys = false;
+    let mut change_master_keys = false;
+
+    match test_data_type {
+        TestDataType::AllRandomFirstCert => {
+            valid_sigs = rng.gen_range(1..=max_pks);
+            threshold = rng.gen_range(0..=valid_sigs);
+            is_first_certificate = true;
+        }
+        TestDataType::AllRandomNextCert => {
+            valid_sigs = rng.gen_range(1..=max_pks);
+            threshold = rng.gen_range(0..=valid_sigs);
+            is_first_certificate = false;
+        }
+        TestDataType::ValidSigEqThresholdNextCert => {
+            valid_sigs = rng.gen_range(1..=max_pks);
+            threshold = valid_sigs;
+            is_first_certificate = false;
+        }
+        TestDataType::ValidSigGreaterThresholdNextCert => {
+            valid_sigs = rng.gen_range(1..=max_pks);
+            threshold = rng.gen_range(0..valid_sigs);
+            is_first_certificate = false;
+        }
+        TestDataType::ValidSigSmallerThresholdNextCert => {
+            valid_sigs = rng.gen_range(1..max_pks);
+            threshold = rng.gen_range(valid_sigs + 1..=max_pks);
+            is_first_certificate = false;
+        }
+        TestDataType::ValidSigAndThresholdEqZeroNextCert => {
+            valid_sigs = 0;
+            threshold = 0;
+            is_first_certificate = false;
+        }
+        TestDataType::ValidSigAndThresholdEqValidatorsNextCert => {
+            valid_sigs = max_pks;
+            threshold = max_pks;
+            is_first_certificate = false;
+        }
+        TestDataType::ValidSigEqValidatorsThresholdEqZeroNextCert => {
+            valid_sigs = max_pks;
+            threshold = 0;
+            is_first_certificate = false;
+        }
+        TestDataType::ValidSigEqZeroThresholdEqValidatorsNextCert => {
+            valid_sigs = 0;
+            threshold = max_pks;
+            is_first_certificate = false;
+        }
+        TestDataType::ChangeMasterKeysNextCert => {
+            valid_sigs = rng.gen_range(1..=max_pks);
+            threshold = rng.gen_range(0..=valid_sigs);
+            is_first_certificate = false;
+            change_signing_keys = false;
+            change_master_keys = true;
+        }
+        TestDataType::ChangeSigningAndMasterKeysNextCert => {
+            valid_sigs = rng.gen_range(1..=max_pks);
+            threshold = rng.gen_range(0..=valid_sigs);
+            is_first_certificate = false;
+            change_signing_keys = true;
+            change_master_keys = true;
+        }
+    }
+    let expected_threshold_result = valid_sigs >= threshold;
+
+    (
+        max_pks,
+        num_custom_fields,
+        valid_sigs,
+        threshold,
+        is_first_certificate,
+        expected_threshold_result,
+        change_signing_keys,
+        change_master_keys,
+    )
+}
+
+fn get_test_circuit_instance(
```

Comment:
I think that this method can be shortened by calling the utilities functions defined in `utils.rs`